lib/utils/isStandardSyntaxRule.js

```diff
@@ -37,6 +37,11 @@ module.exports = function (rule/*: Object*/)/*: boolean*/ {
     return false
   }
 
+  // Less guards
+  if ((/when.*?\(.*?\)/).test(selector)) {
```

Comment:
I think the regexp should account for this limited scenario, instead of allowing for any arbitrary characters between `when` and `(`.